covid/question_covid.py

```diff
@@ -0,0 +1,114 @@
+from transformers import BertTokenizer, BertForQuestionAnswering
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics.pairwise import cosine_similarity
+
+import pandas as pd
+import torch
+
+import json
+import os
+from argparse import ArgumentParser
+
+class QuestionCovid:
+
+    def __init__(
+            self,
+            TOKENIZER,
+            MODEL
+            ):
+        self.TOKENIZER = TOKENIZER
+        self.MODEL = MODEL
+
+    def fit(self, data_text):
+
+        self.TFIDF_VECTORIZER = TfidfVectorizer()
+        self.ARTICLES_MATRIX = self.TFIDF_VECTORIZER.fit_transform(data_text.values())
+
+    def get_answer(self, text, question):
+
+        input_text = "[CLS] " + question + " [SEP] " + text + " [SEP]"
+        input_ids = self.TOKENIZER.encode(input_text)
+        token_type_ids = [0 if i <= input_ids.index(102) else 1 for i in range(len(input_ids))]
+        start_scores, end_scores = self.MODEL(torch.tensor([input_ids]), token_type_ids=torch.tensor([token_type_ids]))
+        all_tokens = self.TOKENIZER.convert_ids_to_tokens(input_ids)
+        answer = ' '.join(all_tokens[torch.argmax(start_scores) : torch.argmax(end_scores)+1])
+
+        return answer, start_scores.max(), end_scores.max()
+
+    def predict(self, question, data_text, index2paperID):
+
+        query = self.TFIDF_VECTORIZER.transform([question])
+        best_matches = sorted([(i,c) for i, c in enumerate(cosine_similarity(query, self.ARTICLES_MATRIX).ravel())], key=lambda x: x[1], reverse=True)
+        best_score = 0 # if score is negative, i consider the answer wrong
+        best_answer = None
+        best_text = None
+        for i, _ in best_matches[:5]:
+            print(i)
```

Comment:
it would be nice to print `index2paperID[i]` instead of i here so that we can inspect the paper. will amend and merge PR